node/impl/full/f3.go

```diff
@@ -0,0 +1,60 @@
+package full
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/fx"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-f3/certs"
+
+	"github.com/filecoin-project/lotus/chain/lf3"
+)
+
+type F3API struct {
+	fx.In
+
+	F3 *lf3.F3 `optional:"true"`
+}
+
+var ErrF3Disabled = errors.New("F3 is disabled")
+
+func (f3api *F3API) F3Participate(ctx context.Context, miner address.Address) (<-chan string, error) {
+	// make channel with some buffere to avoid blocking under higher load
+	errCh := make(chan string, 4)
+
+	if f3api.F3 == nil {
+		log.Infof("F3Participate called for %v, F3 is disabled", miner)
+		// we return a channel that will never be closed
```

Comment:
why not just error so they know it's not possible?

Reply:
I think this is because the implementation treats any error returned as the second return type as retriable, except context cancellation. See [here](https://github.com/filecoin-project/lotus/blob/9a7f0f27b82e0e3f96447f96a1147b44e9836d76/node/modules/storageminer.go#L373). 

Returning a never-closed channel affectively signals to downstream to "do nothing". This is a bit convoluted considering `ErrF3Disabled` exists. 

Assuming I have not missed anything I will go ahead and make this change. Cc @Kubuxu